modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceReleaser.java

```diff
@@ -188,17 +195,60 @@ private boolean hasDeclaredMethod(Class<?> clazz, String methodName, Class<?>...
    */
   private void shutdownMySqlAbandonedConnectionCleanupThread() {
 
-    Class<?> classAbandonedConnectionCleanupThread;
+    Class<?> classAbandonedConnectionCleanupThread = null;
     try {
-      classAbandonedConnectionCleanupThread =
-          this.getClass().getClassLoader().loadClass("com.mysql.jdbc.AbandonedConnectionCleanupThread");
+      for (String knownCleanupThreadClassAddress : CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES) {
+        try {
+          classAbandonedConnectionCleanupThread =
+              this.getClass().getClassLoader().loadClass(knownCleanupThreadClassAddress);
+        } catch (ClassNotFoundException e) {
+          logger.warn("No AbandonedConnectionCleanupThread registered with class address " + knownCleanupThreadClassAddress);
+          continue;
+        }
+      }
+      if (classAbandonedConnectionCleanupThread == null) {
+        logger.warn("No AbandonedConnectionCleanupThread class found. Check if thread class is between "
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Invoke method(s) only conditionally. Use the built-in formatting to construct this argument. ([squid:S2629](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS2629))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7685&issues=AWjzLS16otIXnmsUYkAp&open=AWjzLS16otIXnmsUYkAp)

[ISSUE_KEY]: AWjzLS16otIXnmsUYkAp